### PR TITLE
Refactor/XRPL connector 25 Feb Improvements

### PR DESCRIPTION
**Pull request by mlguys**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
- Improve market info loading logic

**Tests performed by the developer**:
- Run example scripts for XRPL connector in script folder

**Tips for QA testing**:
- This should be tested and merged with its counterpart PR in gateway: https://github.com/hummingbot/gateway/pull/286

### CODE REVIEW

**Review by rapcmia — APPROVED** [review]

File: hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py [hunk 1]
```diff
@@ -80,6 +84,37 @@ async def start(self):
     async def stop(self):
         await super().stop()
 
+    async def _update_markets(self):
+        if self._is_loading_markets:
```
Comment (cardosofede) [inline hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py:88]: is better to use asyncio.Event() to manage this, I think that the logic can be improved here.

**Review by cardosofede — CHANGES_REQUESTED** [review]

File: hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py (hunk 1 above)
Comment (mlguys) [reply hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py]: Thanks, replaced that with asyncio.Event() 👍

**Review by cardosofede — APPROVED** [review]
As we discussed, this behavior should be the same as using the async. Here an example of the two behaviors: https://gist.github.com/cardosofede/c9138731c04036807e7762efcc9be479

File: hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py [hunk 2]
```diff
@@ -80,6 +85,40 @@ async def start(self):
...
+            for market_info in await self._get_markets_info():
+                trading_pair = self._get_trading_pair_from_market_info(market_info=market_info)
+                self._markets_info[trading_pair] = market_info
+                base, quote = split_hb_trading_pair(trading_pair=trading_pair)
+                base_exchange, quote_exchange = self._get_exchange_base_quote_tokens_from_market_info(
+                    market_info=market_info
+                )
+                self._hb_to_exchange_tokens_map[base] = base_exchange
+                self._hb_to_exchange_tokens_map[quote] = quote_exchange
+        finally:
```
Comment (cardosofede) [inline hummingbot/connector/gateway/clob_spot/data_sources/xrpl/xrpl_api_data_source.py:107]: probably adding a except and logging the error will be good to understand what is happening

**Review by cardosofede — CHANGES_REQUESTED** [review]
As we discussed, this behavior should be the same as using the async. Here an example of the two behaviors: https://gist.github.com/cardosofede/c9138731c04036807e7762efcc9be479

**Review by cardosofede — COMMENTED** [review]
As we discussed, this behavior should be the same as using the async. Here an example of the two behaviors: https://gist.github.com/cardosofede/c9138731c04036807e7762efcc9be479

**Review by cardosofede — COMMENTED** [review]
As we discussed, this behavior should be the same as using the async. Here an example of the two behaviors: https://gist.github.com/cardosofede/c9138731c04036807e7762efcc9be479

**Comment by mlguys** [comment]
Hey, I'm closing this PR since the changes are now irrelevant to merge.

**Pull request closed without merging**